module/data/activity/attack-data.mjs

```diff
@@ -329,6 +348,40 @@ export default class AttackActivityData extends BaseActivityData {
 
   /* -------------------------------------------- */
 
+  /**
+   * Create a label based on this activity's settings and, if contained in a weapon, additional details from the weapon.
+   * @returns {string}
+   */
+  getRangeLabel() {
+    if ( this.item.type !== "weapon" ) return this.labels?.range ?? "";
+
+    // TODO: Use proper unit formatting with these once https://github.com/foundryvtt/dnd5e/issues/3958 is resolved
+
+    const parts = [];
+
+    // Add reach for melee weapons, unless the activity is explicitly specified as a ranged attack
+    if ( this.validAttackTypes.has("melee") ) {
+      const { reach, units } = this.item.system.range;
+      parts.push(game.i18n.format("DND5E.RANGE.Formatted.Reach", { reach: `${reach ?? 5} ${units}` }));
+    }
+
+    // Add range for ranged or thrown weapons, unless the activity is explicitly specified as melee
+    if ( this.validAttackTypes.has("ranged") ) {
+      let range;
+      if ( this.range.override ) range = `${this.range.value} ${this.range.units ?? ""}`;
+      else {
+        const { value, long, units } = this.item.system.range;
+        if ( value !== long ) range = `${value}/${long} ${units}`;
```

Comment:
We should check there is actually a value for `long`. I know it's required to have both increments for a ranged or thrown weapon, but the form doesn't enforce it.

```suggestion
        if ( long && (value !== long) ) range = `${value}/${long} ${units}`;
```